quickwit/quickwit-cluster/src/cluster.rs

```diff
@@ -559,11 +594,160 @@ mod tests {
             HashSet::from_iter([QuickwitService::Indexer, QuickwitService::Metastore].into_iter())
         );
         assert_eq!(
-            member_node_2.running_indexing_plan,
-            Some(running_indexing_plan)
+            member_node_2.indexing_tasks,
+            vec![indexing_task.clone(), indexing_task.clone()]
         );
     }
 
+    #[tokio::test]
+    async fn test_chitchat_state_set_high_number_of_tasks() {
+        let transport = ChannelTransport::default();
+        let cluster1 = create_cluster_for_test(Vec::new(), &["indexer"], &transport, true)
+            .await
+            .unwrap();
+        let cluster2 = Arc::new(
+            create_cluster_for_test(
+                vec![cluster1.gossip_listen_addr.to_string()],
+                &["indexer", "metastore"],
+                &transport,
+                true,
+            )
+            .await
+            .unwrap(),
+        );
+        let cluster3 = Arc::new(
+            create_cluster_for_test(
+                vec![cluster1.gossip_listen_addr.to_string()],
+                &["indexer", "metastore"],
+                &transport,
+                true,
+            )
+            .await
+            .unwrap(),
+        );
+        let mut random_generator = rand::thread_rng();
+        let indexing_tasks = (0..100_000)
```

Comment:
The test is failing here, I will either reduce the number of tasks or increase the timeout.